salt/modules/neutronng.py

```diff
@@ -0,0 +1,484 @@
+# -*- coding: utf-8 -*-
+'''
+Neutron module for interacting with OpenStack Neutron
+
+.. versionadded:: Nitrogen
+
+:depends:shade
+
+Example configuration
+
+.. code-block:: yaml
+    neutron:
+      cloud: default
+
+.. code-block:: yaml
+    neutron:
+      auth:
+        username: admin
+        password: password123
+        user_domain_name: mydomain
+        project_name: myproject
+        project_domain_name: myproject
+        auth_url: https://example.org:5000/v3
+      identity_api_version: 3
+'''
+
+from __future__ import absolute_import
+
+import salt.utils
+
+HAS_SHADE = False
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    pass
+
+__virtualname__ = 'neutronng'
+
+
+def __virtual__():
+    '''
+    Only load this module if shade python module is installed
+    '''
+    if HAS_SHADE:
+        return __virtualname__
+    return (False, 'The neutronng execution module failed to \
+                    load: shade python module is not available')
+
+
+def compare_changes(obj, **kwargs):
+    '''
+    Compare two dicts returning only keys that exist in the first dict and are
+    different in the second one
+    '''
+    changes = {}
+    for key, value in obj.items():
+        if key in kwargs:
+            if value != kwargs[key]:
+                changes[key] = kwargs[key]
+    return changes
+
+
+def _clean_kwargs(keep_name=False, **kwargs):
+    '''
+    Sanatize the the arguments for use with shade
+    '''
+    if 'name' in kwargs and not keep_name:
+        kwargs['name_or_id'] = kwargs.pop('name')
+
+    try:
```

Comment:
were going to go ahead and remove this. see https://github.com/saltstack/salt/pull/44817/commits/43c68047c5440793ae59647d1b7de1e76c581695